Firestore/core/src/firebase/firestore/model/no_document.cc

```diff
@@ -22,8 +22,11 @@ namespace firebase {
 namespace firestore {
 namespace model {
 
-NoDocument::NoDocument(DocumentKey key, SnapshotVersion version)
-    : MaybeDocument(std::move(key), std::move(version)) {
+NoDocument::NoDocument(DocumentKey key,
+                       SnapshotVersion version,
+                       bool has_committed_mutations)
+    : MaybeDocument(std::move(key), std::move(version)),
```

Comment:
Optional nit: moving `SnapshotVersion` is no more efficient than copying it (unless there's an expectation this class will grow) (here and in some other constructors)

Reply:
For non-primitive types, I usually just std::move it, unless it's *really* obvious that the thing is small. (really obvious => documented as small.) We can probably assume that with SnapshotVersion (and Timestamp), since it seems unlikely that either will grow.

Somewhat related: http://go/c-readability-advice#value-param

Proposal: Since you've called this optional, and since we std::move SnapshotVersion's (and probably Timestamp's too) somewhat inconsistently in the rest of the code, I'll defer this for now. But I'll create a followup inspired by the above readability link that:
a) ensures std::is_trivially_copyable()
b) ensures sizeof <= 16
for these types, and then:
c) remove usages of std::move and const&.